controllers/eventing/clustercloudeventsource_controller.go

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright 2024 The KEDA Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package eventing
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/tools/cache"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	eventingv1alpha1 "github.com/kedacore/keda/v2/apis/eventing/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/eventemitter"
+	"github.com/kedacore/keda/v2/pkg/metricscollector"
+	kedastatus "github.com/kedacore/keda/v2/pkg/status"
+	"github.com/kedacore/keda/v2/pkg/util"
+)
+
+// ClusterCloudEventSourceReconciler reconciles a EventSource object
+type ClusterCloudEventSourceReconciler struct {
+	client.Client
+	eventEmitter eventemitter.EventHandler
+
+	clusterCloudEventSourceGenerations *sync.Map
+	eventSourcePromMetricsMap          map[string]string
+	eventSourcePromMetricsLock         *sync.Mutex
+}
+
+// NewClusterCloudEventSourceReconciler creates a new ClusterCloudEventSourceReconciler
+func NewClusterCloudEventSourceReconciler(c client.Client, e eventemitter.EventHandler) *ClusterCloudEventSourceReconciler {
+	return &ClusterCloudEventSourceReconciler{
+		Client:                             c,
+		eventEmitter:                       e,
+		clusterCloudEventSourceGenerations: &sync.Map{},
+		eventSourcePromMetricsMap:          make(map[string]string),
+		eventSourcePromMetricsLock:         &sync.Mutex{},
+	}
+}
+
+// +kubebuilder:rbac:groups=eventing.keda.sh,resources=clustercloudeventsources;clustercloudeventsources/status,verbs="*"
+
+// Reconcile performs reconciliation on the identified EventSource resource based on the request information passed, returns the result and an error (if any).
+//
+//nolint:dupl
```

Comment:
I did a quick diff with `CloudEventSourceReconciler` and these two seem to be identical, I was wondering if we could reuse the common code rather than duplicate it.